### ISSUE TITLE
AR Adjustment doesn't support custom fields

### ISSUE THREAD

**Issue by okmotsan** [intacct/intacct-sdk-net#203]
There appears to be a contradiction.  In the ArAdjustmentCreate class in the WriteXml function it outputs the CustomFields into the XML data.  However, in the API docs on the Intacct website there doesn't appear to be support for custom fields for the create_aradjustment XML API call and this seems to be true as when I send the request from .NET with custom fields set for an ArAdjustmentCreate object then I get an error indicating that the custom field node is not expected.

However in Intacct under the Platform Services it does allow me to create custom fields for the AR Adjustment object.

Also the counterpart in the payables (AP Adjustment) does appear to support custom fields.

TL;DR; why doesn't the AR Adjustment support custom fields via the API?  It looks partially implemented but broken
![custom-fields-code](https://user-images.githubusercontent.com/107003259/172246241-013ee4c8-f949-4459-802f-1e115a408a1e.png)
![intacct-custom](https://user-images.githubusercontent.com/107003259/172246243-10524b5f-0780-4251-ac14-9e5102c23b7a.png)
![AP-adj](https://user-images.githubusercontent.com/107003259/172246244-91dcdd9c-f206-4871-bcbd-8ab967825bdc.png)
![AR-missing](https://user-images.githubusercontent.com/107003259/172246245-65f2e355-e056-4563-814b-70e9bca69060.png)

**Comment by dylan-holt-sage** [comment]
@okmotsan, interesting find...... we are going to look into this and get this resolved.... I'll keep you posted.

**Comment by dylan-holt-sage** [comment]
@okmotsan, can you share a snippet or example payload with sensitive information removed? We've researched internally and would like to verify how you're sending.

**Comment by okmotsan** [comment]
`<?xml version="1.0" encoding="utf-8"?>
<function controlid="6435d077-b035-4f76-83af-3590429bd386">
  <create_aradjustment>
    <customerid>C01566</customerid>
    <datecreated>
      <year>2022</year>
      <month>09</month>
      <day>19</day>
    </datecreated>
    <adjustmentno>15549</adjustmentno>
    <invoiceno>1325021</invoiceno>
    <description>Quality</description>
    <customfields>
      <customfield>
        <customfieldname>FRONT_END_SYSTEM</customfieldname>
        <customfieldvalue>RedactedName</customfieldvalue>
      </customfield>
    </customfields>
    <aradjustmentitems>
      <lineitem>
        <glaccountno>21340</glaccountno>
        <amount>-0.05</amount>
        <locationid>REPP</locationid>
        <customerid>C01566</customerid>
      </lineitem>
      <lineitem>
        <glaccountno>40250</glaccountno>
        <amount>-1</amount>
        <locationid>REPP</locationid>
        <customerid>C01566</customerid>
        <itemid/>
      </lineitem>
    </aradjustmentitems>
  </create_aradjustment>
</function>`

I get the response:
Response control status failure - XL03000003 XML Parse schema error: Error 1871: Element 'customfields': This element is not expected. Expected is one of ( externalid, basecurr, currency, nogl, taxsolutionid, aradjustmentitems ).. Line: 1, column: 0. [Support ID: rZQnrEB036%7EYyi3TP0U2TA-drwWK2JRHQAAAAo]

**Comment by okmotsan** [comment]
Is there any ETA for when this can be fixed?  For our use case we have custom fields on both the Invoice and Credit (AR Adjustment) but only the Invoice appears to support populating the customer fields via the API

**Comment by dylan-holt-sage** [comment]
@okmotsan, waiting for a response on a question I have from the team responsible on this. Based on their response, I'll let you know what can be done, if anything.

**Comment by liamstannard** [comment]
Was this ever fixed?

**Comment by Ansh-S** [comment]
@dylan-holt-sage,  was this fixed ? I am having the same trouble with BillCreate object. The SDK lets me pass the custom fields and returns a status of success as well but while retrieving back that bill record, I don't get the custom fields in the response.

**Comment by ericreid10** [comment]
This still appears to be broken in version 3.2.2 of SDK, which at the time of the post is the latest version of the SDK.
